tests/ga/test_update.py

```diff
@@ -1479,6 +1486,118 @@ def test_telemetry_heartbeat_creates_event(self, patch_add_event, patch_info, *_
+            last_modification_time = os.path.getmtime(handler_env_file)
+            self.assertEqual(os.path.getctime(handler_env_file), last_modification_time,
+                             "The creation time and last modified time of the HandlerEnvironment file dont match")
+
+            # Rerun the update handler and ensure that the HandlerEnvironment file is recreated with eventsFolder
+            # flag in HandlerEnvironment.json file
+            with patch('azurelinuxagent.ga.exthandlers._ENABLE_EXTENSION_TELEMETRY_PIPELINE',
+                       return_value=True):
+                update_handler.set_iterations(1)
+                update_handler.run(debug=True)
+
+            self.assertGreater(os.path.getmtime(handler_env_file), last_modification_time,
+                                "HandlerEnvironment file didn't get overwritten")
+
+            with open(handler_env_file, 'r') as handler_env_content_file:
+                content = json.load(handler_env_content_file)
+            self.assertIn(HandlerEnvironment.eventsFolder, content[0][HandlerEnvironment.handlerEnvironment],
+                          "{0} not found in HandlerEnv file".format(HandlerEnvironment.eventsFolder))
+
+    @contextlib.contextmanager
+    def _setup_test_for_ext_event_dirs_retention(self):
+        with self._get_update_handler(test_data=DATA_FILE_MULTIPLE_EXT) as (update_handler, protocol):
+            with patch('azurelinuxagent.ga.exthandlers._ENABLE_EXTENSION_TELEMETRY_PIPELINE', True):
+                update_handler.run(debug=True)
+                expected_events_dirs = glob.glob(os.path.join(conf.get_ext_log_dir(), "*", EVENTS_DIRECTORY))
+                no_of_extensions = protocol.mock_wire_data.get_no_of_plugins_in_extension_config()
+                # Ensure extensions installed and events directory created
+                self.assertEqual(len(expected_events_dirs), no_of_extensions, "Extension events directories dont match")
+                for ext_dir in expected_events_dirs:
+                    self.assertTrue(os.path.exists(ext_dir), "Extension directory {0} not created!".format(ext_dir))
+
+                yield update_handler, expected_events_dirs
+
+    def test_it_should_delete_extension_events_directory_if_extension_telemetry_pipeline_disabled(self):
+
+            # Disable extension telemetry pipeline and ensure events directory got deleted
+            with self._setup_test_for_ext_event_dirs_retention() as (update_handler, expected_events_dirs):
+                with patch('azurelinuxagent.ga.exthandlers._ENABLE_EXTENSION_TELEMETRY_PIPELINE', False):
+                    update_handler.run(debug=True)
+                    for ext_dir in expected_events_dirs:
+                        self.assertFalse(os.path.exists(ext_dir), "Extension directory {0} still exists!".format(ext_dir))
+
+    def test_it_should_retain_events_directories_if_extension_telemetry_pipeline_enabled(self):
```

Comment:
nit: "extension events" instead of just "events" in the method name

Reply:
Done